independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationLiteralProcessor.java

```diff
@@ -82,8 +82,10 @@ public ResultHandle create(BytecodeCreator bytecode, ClassInfo annotationClass,
         if (!annotationInstance.runtimeVisible()) {
             throw new IllegalArgumentException("Annotation does not have @Retention(RUNTIME): " + annotationInstance);
         }
+        if (annotationClass == null) {
+            throw new IllegalArgumentException("Annotation class not available: " + annotationInstance);
+        }
 
-        Objects.requireNonNull(annotationClass, "Annotation class not available: " + annotationInstance);
```

Comment:
`AnnotationInstance#toString()` has a cost so better not pay it when we are in the happy case.